test/gui/shared/scripts/bdd_hooks.py

```diff
@@ -127,10 +128,14 @@ def hook(context):
         filename = context._data["title"].replace(" ", "_") + ".png"
         directory = os.environ["GUI_TEST_REPORT_DIR"] + "/screenshots"
 
-        if not os.path.exists(directory):
+        while not (os.path.exists(directory)):
             os.makedirs(directory)
```

Comment:
You might get infinite loop (in rare case). `if` is good to go.
```suggestion
        if not os.path.exists(directory):
            os.makedirs(directory)
```